### DISCUSSION TITLE
Nextcloudpi script fails on setup.

### DISCUSSION THREAD

**Discussion by dhettinger** [tteck/Proxmox#3908] [Help Needed]
Good day. I am installing the Nextcloud via the nextcloudpi script. Everything goes well through the config but I am met with a curl error. Please see below.

Config settings:
>Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 0
Using Root Password: ********
Container ID: 170
Using Hostname: nextcloudpi
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: no
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: yes
Creating a NextCloudPi LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-zfs for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 170 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 10.10.10.128 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.116.3

verbose error:
>✓ Installed Dependencies
Reading package lists... Donetience)   
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libnss-myhostname libnss-resolve
Suggested packages:
  polkitd | policykit-1
 \The following NEW packages will be installed:
  libnss-myhostname libnss-resolve systemd-resolved
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 490 kB of archives.
After this operation, 1,250 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bookworm/main amd64 libnss-myhostname amd64 252.30-1~deb12u2 [84.7 kB]
Get:2 http://deb.debian.org/debian bookworm/main amd64 systemd-resolved amd64 252.30-1~deb12u2 [307 kB]
Get:3 http://deb.debian.org/debian bookworm/main amd64 libnss-resolve amd64 252.30-1~deb12u2 [97.9 kB]
Fetched 490 kB in 0s (20.9 MB/s)         
 -Selecting previously unselected package libnss-myhostname:amd64.
(Reading database ... 20058 files and directories currently installed.)
Preparing to unpack .../libnss-myhostname_252.30-1~deb12u2_amd64.deb ...
Unpacking libnss-myhostname:amd64 (252.30-1~deb12u2) ...
Selecting previously unselected package systemd-resolved.
Preparing to unpack .../systemd-resolved_252.30-1~deb12u2_amd64.deb ...
Unpacking systemd-resolved (252.30-1~deb12u2) ...
 \Selecting previously unselected package libnss-resolve:amd64.
Preparing to unpack .../libnss-resolve_252.30-1~deb12u2_amd64.deb ...
Unpacking libnss-resolve:amd64 (252.30-1~deb12u2) ...
Setting up libnss-myhostname:amd64 (252.30-1~deb12u2) ...
Setting up systemd-resolved (252.30-1~deb12u2) ...
Converting /etc/resolv.conf to a symlink to /run/systemd/resolve/stub-resolv.conf...
Creating group 'systemd-resolve' with GID 996.
Creating user 'systemd-resolve' (systemd Resolver) with UID 996 and GID 996.
 |Created symlink /etc/systemd/system/dbus-org.freedesktop.resolve1.service → /lib/systemd/system/systemd-resolved.service.
Created symlink /etc/systemd/system/sysinit.target.wants/systemd-resolved.service → /lib/systemd/system/systemd-resolved.service.
 \Setting up libnss-resolve:amd64 (252.30-1~deb12u2) ...
Processing triggers for man-db (2.11.2-2) ...
 \Processing triggers for dbus (1.14.10-1~deb12u1) ...
Processing triggers for libc-bin (2.36-9+deb12u8) ...
 \curl: (6) Could not resolve host: raw.githubusercontent.com
/dev/fd/63: line 1: $'\E[?25h': command not found
/dev/fd/63: line 2: $'\E[01': command not found
/dev/fd/63: line 2: $'31m[ERROR]\E[m': command not found
/dev/fd/63: line 2: $'31m31\E[m:': command not found
/dev/fd/63: line 2: $'31m0\E[m:': command not found

>[ERROR] in line 31: exit code 0: while executing command $STD bash <(curl -fsSL https://raw.githubusercontent.com/nextcloud/nextcloudpi/master/install.sh)

Thank you for looking. Excited to have this resolved. Cheers.

**Comment by tteck** [comment]
If you click on the above link what do you see?
It states `Could not resolve host: raw.githubusercontent.com`.

**Reply by dhettinger** [reply]
Locally raw.githubusercontent.com resolves to github successfully, and I have been able to pull many of your scripts from the raw.hithubusercontent.com domain.

> https://raw.githubusercontent.com/nextcloud/nextcloudpi/master/install.sh

links to the script without error, I was quick to check DNS before posting and local seems to be fine.

I've also attempted a install with the default options and am met with the same error.

--
I did the setup again with my above advanced config resulting in the same error. I was able to log in and there does seem to be some DNS issues, while I am able to ping 185.199.111.133 without issue github.com (or any other address) fails to resolve.

resolvectl status :
>Global
          Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported
   resolv.conf mode: stub
 Current DNS Server: 8.8.8.8
         DNS Servers 8.8.8.8
Fallback DNS Servers 8.8.4.4

>Link 2 (eth0)
Current Scopes: LLMNR/IPv4 LLMNR/IPv6
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

Not really sure how to advance from here, thank you for your support and the amazing scripts. =)

**Reply by tteck** [reply]
what's the output from `cat /etc/resolv.conf`?

**Reply by dhettinger** [reply]
> domain (domain).me
search (domain).me
nameserver 10.10.10.1

which is my local DNS
